### ISSUE TITLE
Downloading inventory fails

### ISSUE THREAD

**Issue by m3nu** [vsespb/mt-aws-glacier#33]
I'm trying to download a vault inventory of about 5000 files, which fails with the following error message:

``` bash
manu@nyx:/Users/manu/…/mt-aws-glacier$./mtglacier download-inventory --config=glacier.cfg --vault=personal --new-journal=personal.log
MT-AWS-Glacier, Copyright 2012-2013 Victor Efimov http://mt-aws.com/ Version 0.942beta

PID 8810 Started worker
PID 8810 Fetched job list for inventory retrieval
PID 8810 Downloaded inventory in JSON format
DIE outside EVAL block [0]
Call stack: 
App::MtAws::main(./mtglacier:5)
App::MtAws::ForkEngine::with_forks(/Users/manu/Downloads/mt-aws-glacier/lib/App/MtAws.pm:371)
App::MtAws::__ANON__(lib/App/MtAws/ForkEngine.pm:54)
App::MtAws::__ANON__(/Users/manu/Downloads/mt-aws-glacier/lib/App/MtAws.pm:348)
Fatal Error: 0 unexpected end of string while parsing JSON string, at character offset 218806 (before "(end of string)") at /Users/manu/Downloads/mt-aws-glacier/lib/App/MtAws.pm line 348, <GEN0> line 2.
```

**Comment by vsespb** [comment]
hi. Do subsequent tries to run this command produce same error message
and stacktrace?

2013/4/23, Manuel Riel notifications@github.com:

**Comment by m3nu** [comment]
Yes. I tried a few times. Can I debug it somehow?

**Comment by vsespb** [comment]
unfortunatelly i can help with debugging only after 25th - i am
travelling now. btw there was similar issue in the past but it self
fixed.

2013/4/23, Manuel Riel notifications@github.com:

**Comment by vsespb** [comment]
Similar to #28

**Comment by vsespb** [comment]
Hello. I am back from vacation, so now can help debug this.

Looks like inventory output from Amazon was truncated 

```
unexpected end of string while parsing JSON string, at character offset 218806 (before "(end of string)")
```

218806 / 5000 files = 43 bytes per file, which is impossible as any archive id is 138 bytes in size.

I expect issue already gone (inventory is expired etc)?

If you see in the future, you can use version 0.943 or higher and issue `mtglacier` with env variable MTGLACIER_DEBUG_INVENTORY=1 set i.e

```
MTGLACIER_DEBUG_INVENTORY=1 mtglacier download-inventory ...
```

this should write two files into current directory:

_download_inventory_body.log
_download_inventory_headers.log

next we need to check that _download_inventory_body contains correct JSON output (no syntax errors in JSON)
you can do it by yourself (using services like http://jsonlint.com/ or software tools) or send to me ( vs at vs-dev dot com ).

(note that both files can contain some private data like SHA256 sums, archive IDs, filenames etc)

If JSON contains syntax errors and `Content-Length` from headers match JSON size, that must be Amazon issue and should be reported to Amazon (together with headers). Otherwise can be some networking issue or even something else.

**Comment by m3nu** [comment]
The error went away after a while, as you predicted. I will keep looking out for it.

**Comment by ghost** [comment]
I seem to be having exactly the same problem as manuelRiel, with the same output.  Unfortunately, the problem hasn't gone away for me after a couple of weeks.  Strangely, two of my inventories were downloaded perfectly, but two terminated with the error above.  I created the files _download_inventory_body.log and _download_inventory_headers.log. Both look ok to me.  Would it be helpful to send them your way?

**Comment by vsespb** [comment]
Yes! vs at vs-dev dot com

**Comment by vsespb** [comment]
For information:
@telscuba sent me inventory output and I found and fixed another bug, not related to this one (bug since 0.944 i.e. introduced 9days ago, fixed in v0.951, description of bug is in ChangeLog https://github.com/vsespb/mt-aws-glacier/blob/master/ChangeLog ).

Issue, reported by @manuelRiel  is different and still open.

**Comment by vsespb** [comment]
Original issue actually could happen due to bug fixed by a064a955b8230d473f420889aac3942a909bf94c in v0.955beta
So closing this. Reopen if happens again.
